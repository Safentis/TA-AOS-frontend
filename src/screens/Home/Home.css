.home {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 10rem 1fr 10rem;
  max-width: 72.2%;
  margin: 0 auto;
  padding: 2rem;
}

.home__filters {
  display: flex;
  justify-self: flex-start;
  justify-content: center;
}

.home__content {
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: 1fr 1fr 1fr;
}

.home__pagination {
  justify-self: center;
  align-self: center;
}

@media screen and (max-width: 768px) {
  .home {
    max-width: 80%;
  }

  .home__content {
    grid-template-columns: 1fr 1fr;
  }
}

@media screen and (max-width: 481px) {
  .home {
    grid-template-rows: 20rem 1fr 10rem;
  }
  
  .home__filters {
    flex-direction: column;
  }
  
  .home__content {
    grid-template-columns: 1fr;
  }
}