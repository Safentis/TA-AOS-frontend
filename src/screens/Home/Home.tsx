import React, { FC, useState } from 'react';
import { Pagination } from 'antd';

import HomeFilters from './HomeFilters/HomeFilters';
import HomeContent from './HomeContent/HomeContent';

import './Home.css';
import { Props } from './Home.interface';
import { useGlobalContext } from '../../App/App.context';

// prettier-ignore
const Home: FC<Props> = ({}): React.ReactElement => {
  const { 
    handlePagination, 
    numberOfPages, 
    numberOfItems,
    currentPage, 
  } = useGlobalContext();

  return (
    <div className="home">
      <HomeFilters className="home__filters" />
      <HomeContent className="home__content" />
      {numberOfPages ? (
        <Pagination
          className="home__pagination"
          current={currentPage}
          total={numberOfItems}
          onChange={handlePagination}
          showSizeChanger={false}
          defaultPageSize={20}
          defaultCurrent={1}
        />) : (
          null
        )
      }
    </div>
  );
};

export default Home;
