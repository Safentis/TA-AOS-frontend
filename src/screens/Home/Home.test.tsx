import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Home from './Home';

import { Props } from './Home.interface';

const setUp = (props: Props) => {
  return shallow(<Home {...props} />);
};

describe('<Home />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {};

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup', () => {
    it('has 3 Label elements', () => {
      expect(component.find('Label').length === 3).toBeTruthy();
    });

    it('has 1 Input element', () => {
      expect(component.find('Input')).toBeTruthy();
    });
  });
});
