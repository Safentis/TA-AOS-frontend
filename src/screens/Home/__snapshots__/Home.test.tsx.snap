// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> component does match to the snapshot 1`] = `
<div
  className="home"
>
  <HomeFilters
    className="home__filters"
  />
  <HomeContent
    className="home__content"
  />
</div>
`;
